### PR TITLE
Matcap

### PR DESCRIPTION
**Pull request by mbotsch**
The PR adds mat-cap textures to `src/pmp/visualization/SurfaceMeshGL`. Since they are a special kind of spherical environment map, the are used in the rendering mode "Texture".

The external repository pmp-data has been updated to include some of Blender's mat-cap files. Find them in `external/pmp-data/matcap/`. 

Currently `mview` is the only application using mat-caps, by specifying `-m matcap-texture.jpg` as command line parameter. We maybe should add a common command line parsing to all applications that derive from MeshViewer...

The PR also improves the parameter passing and file loding in case of an emscripten WebAssembly build. The `mview` web-application now supports to add these parameters in the URL's query part:
- `model=my_model.off`
- `texture=my_texture.jpg`
- `material=my-matcap.jpg`

When the web-app is embedded using an iframe, these attributes can also be added the element attributes to the iframe.

### CODE REVIEW

File: src/pmp/visualization/MatCapShader.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
+//=============================================================================
+// Copyright (C) 2011-2018 The pmp-library developers
```
Comment (dsieger) [inline src/pmp/visualization/MatCapShader.h]: Copyright should be bumped to 2020

File: src/pmp/visualization/PhongShader.h [hunk 2]
```diff
@@ -9,121 +9,130 @@
...
-    "    vec3 rgb;\n"
-    "\n"
-    "    if (use_lighting)\n"
-    "    {\n"
-    "       vec3 L1 = normalize(light1);\n"
-    "       vec3 L2 = normalize(light2);\n"
-    "       vec3 N  = normalize(v2f_normal);\n"
-    "       vec3 V  = normalize(v2f_view);\n"
-    "       \n"
-    //"       if (!gl_FrontFacing) N = -N;\n" // gl_FrontFacing does not work on Mario's new MacBook
```
Comment (dsieger) [inline src/pmp/visualization/PhongShader.h]: Dead code should be removed

File: src/pmp/visualization/SurfaceMeshGL.cpp [hunk 3]
```diff
@@ -40,7 +41,7 @@ SurfaceMeshGL::SurfaceMeshGL()
 
     // material parameters
     front_color_ = vec3(0.6, 0.6, 0.6);
-    back_color_ = vec3(0.5, 0.0, 0.0);
```
Comment (dsieger) [inline src/pmp/visualization/SurfaceMeshGL.cpp]: Why this change? The red rendering of back faces is quite useful, e.g., for detecting intersections or fold-overs.

File: src/pmp/visualization/MatCapShader.h (hunk 1 above)
Comment (mbotsch) [reply src/pmp/visualization/MatCapShader.h]: for new files or for changed files?

File: src/pmp/visualization/PhongShader.h (hunk 2 above)
Comment (mbotsch) [reply src/pmp/visualization/PhongShader.h]: this code will be removed anyway. the new one does not have dead lines.

File: src/pmp/visualization/SurfaceMeshGL.cpp (hunk 3 above)
Comment (mbotsch) [reply src/pmp/visualization/SurfaceMeshGL.cpp]: don't know anymore. something didn't work. let's change it back.

File: src/pmp/visualization/MatCapShader.h (hunk 1 above)
Comment (dsieger) [reply src/pmp/visualization/MatCapShader.h]: Should be updated to current year on every change.

File: src/pmp/visualization/PhongShader.h (hunk 2 above)
Comment (dsieger) [reply src/pmp/visualization/PhongShader.h]: Of course, my bad.

File: src/pmp/visualization/SurfaceMeshGL.cpp (hunk 3 above)
Comment (dsieger) [reply src/pmp/visualization/SurfaceMeshGL.cpp]: Okay, I just pushed a change for that.

File: src/pmp/visualization/MatCapShader.h (hunk 1 above)
Comment (mbotsch) [reply src/pmp/visualization/MatCapShader.h]: updated for all files that changed
